src/jit/codegenarmarch.cpp

```diff
@@ -284,15 +284,21 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 genConsumeAddress(addrNode);
                 addrReg = addrNode->gtRegNum;
 
+                // If addrReg equal to loReg, swap(loReg, hiReg)
+                // This reduces code complexity by only supporting one addrReg overwrite case
+                if (loReg == addrReg)
+                {
+                    loReg = hiReg;
```

Comment:
It will generate an incorrect store sequence, when loReg and hiReg don't contain the lo and hi parts respectively.